lib/charms/mongodb/v1/shards_interface.py

```diff
@@ -272,8 +269,40 @@ def update_mongos_hosts(self):
             self._update_relation_data(relation.id, {HOSTS_KEY: json.dumps(self.charm._unit_ips)})
 
     def get_config_server_status(self):
-        """TODO: Implement this function in a separate PR."""
-        return None
+        """Returns the current status of the config-server."""
+        if not self.charm.is_role(Config.Role.CONFIG_SERVER):
+            logger.info("skipping status check, charm is not running as a shard")
+            return None
+
+        if not self.charm.db_initialised:
+            logger.info("No status for shard to report, waiting for db to be initialised.")
+            return None
+
+        if self.model.relations[LEGACY_REL_NAME]:
+            return BlockedStatus(f"relation {LEGACY_REL_NAME} to shard not supported.")
+
+        if self.model.relations[REL_NAME]:
+            return BlockedStatus(f"relation {REL_NAME} to shard not supported.")
+
+        if not self.is_mongos_running():
+            return BlockedStatus("Internal mongos is not running.")
+
+        shard_draining = self.get_draining_shards()
+        if shard_draining:
+            shard_draining = ",".join(shard_draining)
+            return MaintenanceStatus(f"Draining shard {shard_draining}")
+
+        if not self.model.relations[self.relation_name]:
+            return BlockedStatus("missing relation to shard(s)")
+
+        unreachable_shards = self.get_unreachable_shards()
+        if unreachable_shards:
+            unreachable_shards = "".join(unreachable_shards)
+            return BlockedStatus(f"Shards {unreachable_shards} are unreachable.")
+
+        shards = self.get_related_shards()
+        shards = ", ".join(shards) if len(shards) < 5 else f"{len(shards)} shards"
+        return ActiveStatus(f"config-server connected to {shards}")
```

Comment:
when app is fully started and functioning well, we shouldn't show anything in status. (except "Primary" status for primary)
```suggestion
        return ActiveStatus()
```

Reply:
Ah, okay let me update this (and I will update this for the shard side as well)